substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/code/CodeInfoEncoder.java

```diff
@@ -206,6 +314,83 @@ private void encodeReferenceMaps() {
         }
     }
 
+    public static final int NO_METHOD_METADATA = -1;
+
+    private void encodeMethodMetadata() {
```

Comment:
The sharing is quite important since most strings are method names which are also needed for stack walks. I refactored all the metadata-emitting logic into a new `MethodMetadataEncoder` class marked as `HOSTED_ONLY`.